### ISSUE TITLE
osm mesh upgrade reverts configmap value changes which were made out of band

### ISSUE THREAD

**Issue by michelleN** [openservicemesh/osm#2826]
To reproduce:
```
$ osm install --enable-permissive-traffic-policy=true
$ k get configmap -n osm-system -o yaml
enable_privileged_init_container: "false"
   envoy_log_level: error
   permissive_traffic_policy_mode: "true"
   prometheus_scraping: "true"

$ kubectl patch ConfigMap osm-config -n osm-system -p '{"data":{"permissive_traffic_policy_mode":"false"}}' --type=merge
$ k get configmap -n osm-system -o yaml
enable_privileged_init_container: "false"
   envoy_log_level: error
   permissive_traffic_policy_mode: "false"
   prometheus_scraping: "true"

$ osm mesh upgrade
$ k get configmap -n osm-system -o yaml
apiVersion: v1
envoy_log_level: error
   permissive_traffic_policy_mode: "true"
   prometheus_scraping: "true"
```

Expected ` permissive_traffic_policy_mode: "false"` to not have changed.

**Comment by snehachhabria** [comment]
This will be addressed as a part of #2738

**Comment by snehachhabria** [comment]
We have:
1. replaced the configmap with a meshconfig CRD  
2. have decoupled configuration updates via `mesh upgrade`  cli command 
3. configurations changes will persist through upgrades as `osm-mesh-config` is no longer managed by Helm.

More info on implementation details can be found in this issue : https://github.com/openservicemesh/osm/issues/2738
